### PR TITLE
add functions to add, delete and update todo list tasks in different …

### PR DESCRIPTION
**Pull request by joyapisi**
## What I Have Done🧨 :
`add CRUD (create, read, update, delete) methods`
-Add new folder ./modules is ./src folder
-Add 2 files in ./modules which have the functionalities for adding to local storage and to add, delete and update to do list items

### CODE REVIEW

File: src/modules/add-task.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+function addNewTask(todoTasks, description) {
+  const newTask = {
+    description,
+    completed: false,
+    index: todoTasks.length,
+  };
+  todoTasks.push(newTask);
+}
+
```
Comment (thecodechaser) [inline src/modules/add-task.js]: - [ ] When I'm trying to add a new task, I'm getting the below error. Please make sure to fix this issue and users should be able to add tasks without any errors. Also, make sure other functionalities(remove and edit) work as expected.

image

File: src/index.js [hunk 2]
```diff
@@ -1,33 +1,34 @@
...
+
+let todoTasks = [];
+
+// class TodoTasks {
+//   constructor(description, completed, index) {
+//     this.description = description;
+//     this.completed = completed;
+//     this.index = index;
+//   }
+// }
```
Comment (thecodechaser) [inline src/index.js]: - [x] Please remove the commented code from the files, It's not good practice to leave commented code.

File: src/index.js [hunk 3]
```diff
@@ -40,7 +41,123 @@ function iterateToDoTasks() {
...
+//   const target = event.target;
+//   const item = target.closest('li');
+//   if (target.classList.contains('delete-icon')) {
+//     const index = item.dataset.index;
+//     deleteTask(index);
+//     saveToDoTasks();
+//   }
+// });
+
+// retrieveToDoList();
```
Comment (thecodechaser) [inline src/index.js]: - [x] Please remove the commented code from the files, It's not good practice to leave commented code.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @joyapisi👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- GitHub flow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️

- [x] In the README file please add steps to set up and run the project, It will make the README file more professional. Example: `npm install` `npm run start`.

- [x] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @joyapisi 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - Changes requested have been implemented 👏🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- Kindly update instructions in readme on how to start the app in your setup section like `npm start` 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by joyapisi** [comment]
Add functions to add, delete and update items in Todo list

**Pull request merged**